pkg/crd/testdata/cronjob_types.go

```diff
@@ -297,6 +309,11 @@ type MinMaxObject struct {
 	Baz string `json:"baz,omitempty"`
 }
 
+type EmpiableObject struct {
+	Foo string `json:"foo,omitempty"`
```

Comment:
Can you add a default to the field so that we can prove that the empty default on the parent triggers the inner field to be defaulted.

Reply:
So with a modified version of CRD, where only these 2 new fields are present, this is how the applied CR looks like:
```yaml
kind: CronJob
apiVersion: testdata.kubebuilder.io/v1
metadata:
  name: test
  namespace: default
spec: {}
```
results in
```yaml
apiVersion: testdata.kubebuilder.io/v1
kind: CronJob
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"testdata.kubebuilder.io/v1","kind":"CronJob","metadata":{"annotations":{},"name":"test","namespace":"default"},"spec":{}}
  creationTimestamp: "2023-11-24T16:21:31Z"
  generation: 1
  name: test
  namespace: default
  resourceVersion: "23165"
  uid: d86f95e1-54be-4e04-b789-cb997bc3dd17
spec:
  defaultedEmptyObject:
    foo: forty-two
  defaultedEmptySlice: []
```

Reply:
@JoelSpeed Do you think all the comments are addressed? This is quite useful functionality, and it would be best to avoid manually patching resource, and rely on the annotation instead.